examples/python/trigger-pagerduty-incident/handler/handler.py

```
@@ -0,0 +1,179 @@
+                    'source': cevent['source'],
+                    'severity': 'info',
+                    'component': eventdata['Vm']['Name'],
+                    'group': eventdata['Host']['Name'],
+                    'class': cevent['subject'],
+                    'custom_details': {
+                        'user': eventdata['UserName'],
+                        'Datacenter': eventdata['Datacenter'],
+                        'ComputeResource': eventdata['ComputeResource'],
+                        'Host': eventdata['Host'],
+                        'VM': eventdata['Vm']
+                    }
+                }
+                #,
+                #'images': [{
+                #    'src': 'https://www.pagerduty.com/wp-content/uploads/2016/05/pagerduty-logo-green.png',
+                #    'href': 'https://example.com/',
+                #    'alt': eventdata['Vm']
+                # }],
+                # 'links': [{
+                #    'href': 'https://example.com/',
+                #    'text': 'Link to VM'
+                # }]
+            }
+    except KeyError as err:
+        res = FaaSResponse('400','JSON does not conform to CloudEvent spec > KeyError: {0}'.format(err))
+        print(json.dumps(vars(res)))
+        return
+    except TypeError as err:
+        res = FaaSResponse('400','Error parsing CloudEvent, Unexpected Event received > TypeError: {0}'.format(err))
+        print(json.dumps(vars(res)))
+        return
+
+    #print(obj)
+
+    # Make the Rest Api Call to PagerDuty
+    try:
+        s=requests.Session()
+        if(os.getenv("insecure_ssl")):
+            s.verify=False
+        pg = Pagerduty(s)
+        res = pg.invoke(obj)
+        if res.status != '200':
+            print(json.dumps(vars(res)))
+    except (OSError, KeyError) as err:
+        # this could only occur from the PagerDuty constructor and sufficient logging for these Exceptions are handled there. 
+        return 
```

Since you're handling the exception in the constructor, why not return a proper error message like with `pg.invoke` above and avoid nested exception handling? I.e. handle exceptions in the constructor/invocation and return proper FaaSResponse in this handler? Avoids jumping around in the code to see what's going on, less ambiguity and more clarity to the reader.

---

Updated the exception handling to be more accurate and avoided nested/complex flows